uaclient/clouds/identity.py

```diff
@@ -58,12 +60,20 @@ def get_cloud_type_from_result_file(
     return DATASOURCE_TO_CLOUD_ID.get(dsname, dsname)
 
 
+@lru_cache(maxsize=None)
 @apply_config_settings_override("cloud_type")
 def get_cloud_type() -> "Optional[str]":
+    cfg = UAConfig()
     if util.which("cloud-id"):
         # Present in cloud-init on >= Xenial
-        out, _err = util.subp(["cloud-id"])
-        return out.strip()
+        warning_text = status.NOTICE_CLOUD_ID_ERROR
+        try:
+            out, _err = util.subp(["cloud-id"])
+            cfg.remove_notice("", warning_text)
+            return out.strip()
+        except util.ProcessExecutionError:
+            cfg.add_notice("", warning_text)
+            return "cloud-id-error"
```

Comment:
I think we should return `None`. This is just to make this function still consistent with this part of the [code](https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/clouds/identity.py#L88)

Reply:
Got it. But I think we should somehow separate the cases.
I agree that if there is an error, we could not identify the cloud type when instantiating, but returning `None` and checking it afterwards would trigger the same warning during FIPS installation when we are not running in the cloud.
Maybe there is a better way to diff between the two situations.

Reply:
We should probably return a tuple then, with the cloud-type and reason for failure. Similar to what we do in the `can_enable` method

Reply:
good good, /me on it